src/main.c

```diff
@@ -61,87 +58,58 @@
 // not work.
-static const gchar *ip_address = NULL;
+static const char *ip_address = NULL;
 static int listen_port = 49494;
 
 #ifdef GMRENDER_UUID
 // Compile-time uuid.
-static const gchar *uuid = GMRENDER_UUID;
+static const char *uuid = GMRENDER_UUID;
 #else
-static const gchar *uuid = "GMediaRender-1_0-000-000-002";
+static const char *uuid = "GMediaRender-1_0-000-000-002";
 #endif
-static const gchar *friendly_name = PACKAGE_NAME;
-static const gchar *output = NULL;
-static const gchar *pid_file = NULL;
-static const gchar *log_file = NULL;
-static const gchar *mime_filter = NULL;
+static const char *friendly_name = PACKAGE_NAME;
+static const char *output = NULL;
+static const char *pid_file = NULL;
+static const char *log_file = NULL;
+static const char *mime_filter = NULL;
 
 /* Generic GMediaRender options */
-static GOptionEntry option_entries[] = {
-	{ "version", 0, 0, G_OPTION_ARG_NONE, &show_version,
-	  "Output version information and exit", NULL },
-	{ "ip-address", 'I', 0, G_OPTION_ARG_STRING, &ip_address,
-	  "The local IP address the service is running and advertised "
-	  "(only one, 0.0.0.0 won't work)", NULL },
-	// The following is not very reliable, as libupnp does not set
-	// SO_REUSEADDR by default, so it might increment (sending patch).
-	{ "port", 'p', 0, G_OPTION_ARG_INT, &listen_port,
-	  "Port to listen to; [49152..65535] (libupnp does not use "
-	  "SO_REUSEADDR, so might increment)", NULL },
-	{ "uuid", 'u', 0, G_OPTION_ARG_STRING, &uuid,
-	  "UUID to advertise", NULL },
-	{ "friendly-name", 'f', 0, G_OPTION_ARG_STRING, &friendly_name,
-	  "Friendly name to advertise.", NULL },
-	{ "output", 'o', 0, G_OPTION_ARG_STRING, &output,
-	  "Output module to use.", NULL },
-	{ "pid-file", 'P', 0, G_OPTION_ARG_STRING, &pid_file,
-	  "File the process ID should be written to.", NULL },
-	{ "daemon", 'd', 0, G_OPTION_ARG_NONE, &daemon_mode,
-	  "Run as daemon.", NULL },
-	{ "mime-filter", 0, 0, G_OPTION_ARG_STRING, &mime_filter,
-	  "Filter the supported media types. "
-		"e.g. Audio only: '--mime-filter audio'. Disable FLAC: '--mime-filter -audio/x-flac'.", NULL },
-	{ "logfile", 0, 0, G_OPTION_ARG_STRING, &log_file,
-	  "Debug log filename. Use 'stdout' or 'stderr' to log to console.", NULL },
-	{ "list-outputs", 0, 0, G_OPTION_ARG_NONE, &show_outputs,
-	  "List available output modules and exit", NULL },
-	{ "dump-devicedesc", 0, 0, G_OPTION_ARG_NONE, &show_devicedesc,
-	  "Dump device descriptor XML and exit.", NULL },
-	{ "dump-connmgr-scpd", 0, 0, G_OPTION_ARG_NONE, &show_connmgr_scpd,
-	  "Dump Connection Manager service description XML and exit.", NULL },
-	{ "dump-control-scpd", 0, 0, G_OPTION_ARG_NONE, &show_control_scpd,
-	  "Dump Rendering Control service description XML and exit.", NULL },
-	{ "dump-transport-scpd", 0, 0, G_OPTION_ARG_NONE, &show_transport_scpd,
-	  "Dump A/V Transport service description XML and exit.", NULL },
-	{ NULL }
+static struct option option_entries[] = {
```

Comment:
We're now loosing all the commandline option description provided with `gmediarender -h`; in fact this new version does not have a help anymore.

So we need to get this functionality back somehow differently.